pkg/filter/proxy/proxy.go

```diff
@@ -315,6 +353,20 @@ func (b *Proxy) Handle(ctx context.HTTPContext) (result string) {
 	return ctx.CallNextHandler(result)
 }
 
+func (b *Proxy) updateAndGetClient(tracingInstance *tracing.Tracing) *Client {
+	b.clientMutex.RLock()
+	needsUpdate := b.client.IsTracingUpdated(tracingInstance)
+	b.clientMutex.RUnlock()
+	if !needsUpdate {
+		return b.client
+	}
```

Comment:
Thanks for the idea, updated! However when `tracingInstance` changes, http.Client has to be recreated (`b.createHTTPClient()`). Otherwise if the http.Client is reused, zipkin wrapper hangs for some reason in [DoWithAppSpan](https://github.com/openzipkin/zipkin-go/blob/master/middleware/http/client.go#L111)

Reply:
Cool, thanks for the information.